src/ts/plugins/server/GitlabHostingConnector.ts

```diff
@@ -146,12 +146,32 @@ export default class GitlabHostingConnector extends GitlabConnector implements H
     return `${projectUrl}/pages`
   }
 
-  async getGitlabJobLogsUrl(session: GitlabSession, websiteId: WebsiteId, projectUrl: string): Promise<string> {
-    const jobs = await this.callApi(session, `api/v4/projects/${websiteId}/jobs`, 'GET')
-    return `${projectUrl}/-/jobs/${jobs[0].id}`
+  async getGitlabJobLogsUrl(session: GitlabSession, websiteId: WebsiteId, job: PublicationJobData, { startJob, jobSuccess, jobError }: JobManager, projectUrl: string, tag): Promise<string> {
+    let jobs = await this.callApi(session, `api/v4/projects/${websiteId}/jobs`, 'GET')
```

Comment:
Don't make a first call then many other ones

I suggest :

1. Do
1. await get the tag (callApi)
1. await TIME_BEFORE_RETRY - see [the way to do this (first solution of this article)](https://attacomsian.com/blog/javascript-sleep-delay-wait-pause)
1. While elapsed time < TIMEOUT and the tag is not the expected one

Reply:
"Don't make a first call then many other ones"
I don't understand, I need to get the first time the jobs in order to read its jobs.ref containing the current tag ? otherwise I don't know how to initialize the while loop ?

Reply:
"await TIME_BEFORE_RETRY ", I thought that setTimeout do the same ? https://www.freecodecamp.org/news/javascript-settimeout-how-to-set-a-timer-in-javascript-or-sleep-for-n-seconds/

Reply:
"await get the tag (callApi)
await TIME_BEFORE_RETRY - see [the way to do this (first solution of this article)](https://attacomsian.com/blog/javascript-sleep-delay-wait-pause)
While elapsed time < TIMEOUT and the tag is not the expected one"

I beleived I'm doing exactly this algo ?

Reply:
>> "Don't make a first call then many other ones"

> I don't understand, I need to get the first time the jobs in order to read its jobs.ref containing the current tag ? otherwise I don't know how to initialize the while loop ?

use a do while instead of a while

Reply:
> "await TIME_BEFORE_RETRY ", I thought that setTimeout do the same ? https://www.freecodecamp.org/news/javascript-settimeout-how-to-set-a-timer-in-javascript-or-sleep-for-n-seconds/

the setTimeout takes a callback, and given the complexity of this algorithm you should use await and no callback, it will make it obvious my other feedback

Reply:
thank you for your time to explain to me

Reply:
I will change with do while, thanks for advices